tests/test_ipm.py

```diff
@@ -1,56 +1,85 @@
 import logging
-
 import pytest
+
 from ophyd.sim import make_fake_device
 from unittest.mock import Mock
 
 from pcdsdevices.inout import InOutRecordPositioner
-from pcdsdevices.ipm import IPM, IPMTarget
+from pcdsdevices.ipm import (IPMMotion, IPM_IPIMB, IPM_Wave8, IPMTarget,
+                             IPM, Wave8, IPIMB)
 
 logger = logging.getLogger(__name__)
 
 
-@pytest.fixture(scope='function')
-def fake_ipm():
-    FakeIPM = make_fake_device(IPM)
+def fake_ipm(cls):
+    """
+    Setup a fake ipm, given a real class
+    """
+    FakeIPM = make_fake_device(cls)
     ipm = FakeIPM("Test:My:IPM", name='test_ipm')
-    ipm.diode.state.sim_put(0)
-    ipm.diode.state.sim_set_enum_strs(['Unknown'] +
-                                      InOutRecordPositioner.states_list)
+    ipm.diode.state.state.sim_put(0)
+    ipm.diode.state.state.sim_set_enum_strs(['Unknown'] +
+                                            InOutRecordPositioner.states_list)
     ipm.target.state.sim_put(0)
     ipm.target.state.sim_set_enum_strs(['Unknown'] + IPMTarget.states_list)
     return ipm
 
 
-def test_ipm_states(fake_ipm):
+@pytest.fixture(scope='function')
+def fake_ipm_motion():
+    """
+    Test IPM_Motion
+    """
+    return fake_ipm(IPMMotion)
+
+
+@pytest.fixture(scope='function', params=[IPM_Wave8, IPM_IPIMB])
+def fake_ipm_with_box():
+    """
+    Test IPM_Wave8 and IPM_IPIMB
+    """
+    return fake_ipm(request.param)
```

Comment:
Good concise test class setup :+1:

Reply:
Unfortunately, that way doesn't work so the nice conciseness is gone :(